src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -439,14 +439,13 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
 
                 if (addIndexColumn)
                 {
-                    PrimitiveDataFrameColumn<int> indexColumn = new PrimitiveDataFrameColumn<int>("IndexColumn", columns[0].Length);
```

Comment:
What are your thoughts on the friendly name/aliases? Do you prefer Int64DataFrameColumn over PrimitiveDataFrameColumn< long >?

I'm still trying to figure out the best way to bring back the aliases in the Generic Math branch (or if I should).

Reply:
> Why make the change from PrimitiveDataFrameColimn to Int64DataFrameColumn instead of PrimitiveDataFrameColimn < long >?

At first I just change from PrimitiveDataFrameColumn<int> to PrimitiveDataFrameColumn<long> as DataFrame may use long amount of rows. Second change was from PrimitiveDataFrameColumn<long> to Int64DataFrameColumn - this is due to the issue similar to https://github.com/dotnet/machinelearning/issues/6127

My thoughts in this particular case was that using Int64DataFrameColumn is more safe as it inherits from PrimitiveDataFrameColumn<long> and can be casted to it, so:

```
[Fact]
public void Test()
{
    var index = new Int64DataFrameColumn("index", new int[] { 1, 2, 3, 4, 5 });
    var dfTest = new DataFrame(index);

    var col1 = dfTest.Columns.GetInt64Column("index");
    var col2 = dfTest.Columns.GetPrimitiveColumn<long>("index");
}
```

succeds. In case of using PrimitiveDataFrameColumn<long> it fails.

Reply:
As according to current implementation I think that having both PrimitiveDataFrameColumn<T> and _T_DataFrameColumn is confusing. I'll prefer to have PrimitiveDataFrameColumn<T> as an abstract class, allowing user to provider it's own inheritors for user defined structures (for example Complex numbers). In this case of course PrimitiveDataFrameColumn methods like ToArrowArray, GetDataViewType and so on should be virtual.

Also, in case if we would like to have all DataFrame columns internaly compatible to Apache Arrow format (for non memorycopy conversion), we will get DateDataFrameColumn and DecimalDataFrameColumn not inheriting  from PrimitiveDataFrameColumn<DateTime> and PrimitiveDataFrameColumn<decimal>.

Using Generic Math may change the way how all works and it may allow to eliminate the need in all inheritors of PrimitiveDataFrameColumn. It's a breaking change and that's why may be a good reason for overall DataFrame API redesign in future versions